File: 9.用两个栈实现队列/9.用两个栈实现队列.cpp
class Solution
{
public:
    void push(int node) {
        stack1.push(node);
    }

    int pop() {
        if (stack2.empty()){
            while(!stack1.empty()){
                stack2.push(stack1.top());
                stack1.pop();
            }
        }
        int xx = stack2.top();
        stack2.pop();
        return xx;
    }

private:
    stack<int> stack1;
    stack<int> stack2;
};

File: 9.用两个栈实现队列/9.用两个栈实现队列.py
# -*- coding:utf-8 -*-
class Solution:
    stack1 = []
    stack2 = []
    def push(self, node):
        # write code here
        self.stack1.append(node)
    def pop(self):
        # return xx
        if len(self.stack2)==0:
            while len(self.stack1)!=0:
                self.stack2.append(self.stack1[-1])
                self.stack1.pop()
        xx = self.stack2[-1]
        self.stack2.pop()
        return xx